src/test/java/com/sharewalk/dao/UserDAOImplTest.java

```diff
@@ -0,0 +1,73 @@
+package com.sharewalk.dao;
+
+import com.sharewalk.model.User;
+import com.sharewalk.model.Walk;
+import org.junit.Rule;
+import org.junit.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class UserDAOImplTest {
+    @Rule
+    public MockitoRule mockito = MockitoJUnit.rule();
+
+    @InjectMocks
+    private UserDAOImpl instance;
+
+    @Mock
+    private EntityManager entityManager;
+
+    @Mock
+    Query query;
+
+    @Mock
+    Query query1;
+
+    @Mock
+    Query query2;
+
+    @Test
+    public void listUserWalksWithoutParametersTest(){
+        Long userId = Long.valueOf(1L);
+        when(entityManager.createNamedQuery("Walk.findAllForUser")).thenReturn(query);
+        when(query.setParameter("userid",userId)).thenReturn(query1);
+        when(query1.getResultList()).thenReturn(Arrays.asList(new Walk("walk 1", null, null)));
```

Comment:
If dao does not add any logic, but simply returns list, check if the list is passed, like i wrote in test o controllers